src/ser/mod.rs

```diff
@@ -385,9 +385,9 @@ impl<'a, 'b: 'a> ser::Serializer for &'a mut Serializer<'b> {
     fn serialize_tuple_struct(
         self,
         _name: &'static str,
-        _len: usize,
+        len: usize,
     ) -> Result<Self::SerializeTupleStruct> {
-        unreachable!()
+        self.serialize_seq(Some(len))
```

Comment:
It seems odd to me that we would only support serializing a type, but if you tried to then deserialize the time (that you just serialized), we would internally panic.

I believe that the lack of support for tuple structs is intentional because JSON does not natively have this type, so it can't be easily mapped to/from Rust/Json.

Reply:
That being said, it looks like `serde-json` handles this gracefully, so I see no issue to adding it in.

Can you add support for deserializing these types as well? Also, mind adding  CHANGELOG entry?

Reply:
> Can you add support for deserializing these types as well (if not, we can spawn an issue for it)?

That's the odd thing, deserializing tuple structs was already supported before I worked on this. It seemed odd that *de*serializing them was supported but serializing the same type would panic, so I implemented this. I've added a test just to verify deserializing / round-tripping them also works.

I've also updated the CHANGELOG with this change.

Reply:
That is quite odd indeed! Thanks for the updates then! :)